### PR TITLE
Security

### PR DESCRIPTION
**Pull request by nopbyte**
The time has come  to merge the security version into master :) ... 

There are pull requests for all the containers to merge the security branches that have been modified into master.

To solve the host configuration problem, now os-js and idm are mounting the /etc/hostname file locally and reading it as mentioned by @craig-mulligan in the architecture issue.

Also, the security branch contains the docker-compose file with the security versions that can be tested right away.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
+ * this node-red includes the agile-idm token node to retrieve the token of the currently logged in user.
+* AGILE-IDM
+
+### How to configure it
+
+You need to add something like this to your /etc/hosts file (assuming that the gateway is on 192.168.1.20):
+
+192.168.1.20   agilegw.local
+
+This is needed because the configurations of all the modified components rely in this name (agilegw to find each other; especially, when this comes to Oauth2 integrations). We should address this in the future in the project (see https://github.com/Agile-IoT/Architecture/issues/3)
```
Comment (craig-mulligan) [inline README.md]: Is this the correct issue - https://github.com/Agile-IoT/Architecture/issues/3 ? It doesn't seem relevant?

File: README.md [hunk 2]
```diff
@@ -2,6 +2,37 @@
...
+ * run an express app
+ * get the token from agile-idm in the initial request to the application and verify that only a particular user from IDM can use a node-red instance by mapping the user session to  the admin session from Node-RED
+ * this node-red includes the agile-idm token node to retrieve the token of the currently logged in user.
+* AGILE-IDM
+
+### How to configure it
+
+You need to add something like this to your /etc/hosts file (assuming that the gateway is on 192.168.1.20):
+
+192.168.1.20   agilegw.local
```
Comment (craig-mulligan) [inline README.md]: I thought we were reading the hostname automatically, are there still issues with that?

File: docker-compose.yml [hunk 3]
```diff
@@ -78,6 +80,25 @@ services:
       - agile-core:agile
     restart: always
 
+
+
+  agile-idm:
+    image: agileiot/agile-idm-armv7l
```
Comment (craig-mulligan) [inline docker-compose.yml]: It'd be nice to have versioned tag here but it's okay for now. (same for other images)

File: README.md (hunk 2 above)
Comment (nopbyte) [inline README.md]: You are right, this is not needed any more. I removed it from the README in the latest commit

File: README.md (hunk 1 above)
Comment (nopbyte) [inline README.md]: True. The issue was [https://github.com/Agile-IoT/Architecture/issues/4]. But, this part is also removed from the README now.

File: docker-compose.yml (hunk 3 above)
Comment (nopbyte) [inline docker-compose.yml]: I must say: the automation to build images and upload them to dockerhub was done by @cskiraly  in the agile-idm repo, so in this regard I am clueless.

I guess versionist along with the .travis file can help with this right? If there is a project-wide way to integrate this I would not have an issue with placing this in the security-relevant repos.

File: README.md (hunk 2 above)
Comment (craig-mulligan) [inline README.md]: Great!

File: docker-compose.yml (hunk 3 above)
Comment (craig-mulligan) [inline docker-compose.yml]: Yes, well you'd have to add it for each repo. There's an example here: https://github.com/Agile-IoT/agile-example

If you don't want to use versionist, travis etc. Then you should manually version it, by tagging each docker image before pushing.

**Comment by cskiraly** [comment]
I'm trying a merge. I still see probems with redirection: e.g. "Error: client URL doesn't match what was expected. Provided: http://192.168.8.54:8000/ expected http://agilegw.local:8000/ at /opt/agile-idm-web-ui/routes/oauth2-routes.js:105:23 at /opt/agile-idm-web-ui/lib/db/clients.js:9:12"
Where should I set the IP? Is it documented?

**Comment by nopbyte** [comment]
@cskiraly what happens is that idm takes the content of /etc/hostname and appends ".local" and uses it as the hostname to redirect IDM, register the OSJS application in IDM... etc. In short, this is the name used to redirect the browser to IDM and other interactions requiring the name of the server.

we could change slightly  the /etc/hostname idea, and just edit the hostname file and add the whole name (i.e. with .local at the end or any other fully qualified domain) or the IP that we want to use maybe?

**Pull request merged**